src/QsCompiler/CompilationManager/ProjectManager.cs

```diff
@@ -757,6 +761,36 @@ public ProjectManager(Action<Exception>? exceptionLogger, Action<string, Message
             });
         }
 
+        /// <summary>
+        /// To be used whenever a temporary project file is added.
+        /// </summary>
+        public Task LoadTemporaryProjectAsync(
+            Uri projectFile,
+            ProjectInformation.Loader projectLoader,
+            Func<Uri, FileContentManager?>? openInEditor = null)
+        {
+            openInEditor ??= _ => null;
+
+            return this.load.QueueForExecutionAsync(() =>
+            {
+                if (!projectLoader(projectFile, out var info))
+                {
+                    return;
+                }
+
+                var project = new Project(projectFile, info, this.logException, this.publishDiagnostics, this.log);
+                this.temporaryProjects.AddOrUpdate(projectFile, project, (_, __) => project);
+
+                // If any of the files that are currently open in the editor is part of the project,
+                // then we need to make sure to remove them from the default manager before adding them to the project.
+                project.LoadProjectAsync(
+                    this.temporaryProjects.ToImmutableDictionary(p => p.Key, p => p.Value.OutputPath),
+                    this.MigrateToProject(openInEditor),
```

Comment:
A file is still associated with two projects then, and the issue we ran into with temporary projects is avoided by the split between temp projects and other projects?

Reply:
Edit: answered in our conversation.

Reply:
Yes, that's correct. See the PR description for the rationale of why I chose this route and why I believe it's the best option. Please let me know if you have any questions or disagree with anything there.